src/commonMain/kotlin/fr/acinq/lightning/payment/OutgoingPaymentFailure.kt

```diff
@@ -1,60 +1,111 @@
     fun toPaymentFailure(): OutgoingPaymentFailure = OutgoingPaymentFailure(this, listOf<LightningOutgoingPayment.Part.Status.Failed>())
 
     // @formatter:off
-    object AlreadyPaid : FinalFailure() { override fun toString(): String = "this invoice has already been paid" }
-    object InvalidPaymentAmount : FinalFailure() { override fun toString(): String = "payment amount must be positive" }
-    object FeaturesNotSupported : FinalFailure() { override fun toString(): String = "payment request features not supported" }
-    object InvalidPaymentId : FinalFailure() { override fun toString(): String = "payment ID must be unique" }
-    object NoAvailableChannels : FinalFailure() { override fun toString(): String = "no channels available to send payment" }
-    object InsufficientBalance : FinalFailure() { override fun toString(): String = "not enough funds in wallet to afford payment" }
-    object NoRouteToRecipient : FinalFailure() { override fun toString(): String = "unable to route payment to recipient" }
-    object RecipientUnreachable : FinalFailure() { override fun toString(): String = "the recipient was offline or did not have enough liquidity to receive the payment" }
-    object RetryExhausted: FinalFailure() { override fun toString(): String = "payment attempts exhausted without success" }
-    object WalletRestarted: FinalFailure() { override fun toString(): String = "wallet restarted while a payment was ongoing" }
-    object UnknownError : FinalFailure() { override fun toString(): String = "an unknown error occurred" }
+    data object AlreadyPaid : FinalFailure() { override fun toString(): String = "this invoice has already been paid" }
+    data object InvalidPaymentAmount : FinalFailure() { override fun toString(): String = "payment amount must be positive" }
+    data object FeaturesNotSupported : FinalFailure() { override fun toString(): String = "payment request features not supported" }
+    data object InvalidPaymentId : FinalFailure() { override fun toString(): String = "payment ID must be unique" }
+    data object ChannelNotConnected : FinalFailure() { override fun toString(): String = "channel is not connected yet, please retry when connected" }
+    data object ChannelOpening : FinalFailure() { override fun toString(): String = "channel creation is in progress, please retry when ready" }
+    data object ChannelClosing : FinalFailure() { override fun toString(): String = "channel closing is in progress, please retry when a new channel has been created" }
+    data object NoAvailableChannels : FinalFailure() { override fun toString(): String = "payment could not be sent through existing channels, check individual failures for more details" }
+    data class InsufficientBalance(val available: MilliSatoshi) : FinalFailure() { override fun toString(): String = "not enough funds in wallet to afford payment (available = $available)" }
```

Comment:
Making `InsufficientBalance` a data class is not consistent with the other failures, and brings some complexity down the line for the application. Having the available amount at the time of the payment is not very useful (at least on Phoenix, even if we had that data, we would not display it)

Reply:
Yeah, I wasn't sure whether it was a good idea or not. I removed it in https://github.com/ACINQ/lightning-kmp/pull/634/commits/8bd90a371fde70736b8128eedc6742c8ebe65627